sentry/src/main/java/io/sentry/Span.java

```diff
@@ -163,6 +188,28 @@ public void finish(final @Nullable SpanStatus status, final @Nullable SentryDate
 
     this.context.setStatus(status);
     this.timestamp = timestamp == null ? hub.getOptions().getDateProvider().now() : timestamp;
+    if (options.isTrimStart() || options.isTrimEnd()) {
+      @Nullable SentryDate minChildStart = null;
+      @Nullable SentryDate maxChildEnd = null;
+
+      final @NotNull List<Span> children = getChildren();
+      for (final Span child : children) {
+        if (minChildStart == null || child.getStartDate().isBefore(minChildStart)) {
+          minChildStart = child.getStartDate();
+        }
+        if (maxChildEnd == null
+            || (child.getFinishDate() != null && child.getFinishDate().isAfter(maxChildEnd))) {
+          maxChildEnd = child.getFinishDate();
+        }
+      }
+      if (options.isTrimStart() && minChildStart != null) {
+        updateStartDate(minChildStart);
+      }
+      if (options.isTrimEnd() && maxChildEnd != null) {
+        updateEndDate(maxChildEnd);
```

Comment:
just thinking if we should check here whether the current end timestamp is already less than the `maxChildEnd`? otherwise we're extending and not trimming kinda 😅 shouldn't happen I guess, but just for safety